### PR TITLE
feat: enable dataset access for shared emails

### PR DESCRIPTION
**Pull request by martin-trajanovski**
## Description

This PR aims to make shared datasets available for users. Also prevents all users for being able to update datasets and leave that ability only for owners.

## Motivation

Anyone that has access to dataset could update or share it and people in sharedWith field were not able to access the datasets.

## Fixes:

* https://jira.esss.lu.se/browse/SWAP-3274

## Changes:

* changes made

## Tests included/Docs Updated?

- [x] Included for each change/fix?
- [x] Passing? (Merge will not be approved unless this is checked) 
- [ ] Docs updated?
- [ ] New packages used/requires npm install? 
- [ ] Toggle added for new features?

### CODE REVIEW

File: src/datasets/datasets.controller.ts [hunk 1]
```diff
@@ -941,13 +942,27 @@ export class DatasetsController {
...
+    // NOTE: Not sure if email is the best way to do this but since we don't have userId or something like that email is acceptable.
+    const couldUpdateDataset =
+      datasetToUpdate?.ownerEmail === loggedInUser.email;
     // $addToSet is necessary to append to the field and not overwrite
     // $each is necessary as data is an array of values
 
+    if (!couldUpdateDataset) {
+      throw new ForbiddenException();
+    }
+
```
Comment (nitrosx) [inline src/datasets/datasets.controller.ts:965]: In order to allow the update on the dataset, we should use the casl factory.
The rules is the following:
- if the user logged in belongs to the group that owns the dataset, he/she is allowed to update the dataset.
- if the user belongs to any of the admin groups, he/she can update the dataset.

**Review by nitrosx — CHANGES_REQUESTED** [review]
Please use the casl facility to check if the user can update the dataset.

File: src/datasets/datasets.controller.ts (hunk 1 above)
Comment (martin-trajanovski) [reply src/datasets/datasets.controller.ts:965]: Ahaaa... I see now but do we have these rules set up already or I need to fix that?

Comment (martin-trajanovski) [reply src/datasets/datasets.controller.ts]: @nitrosx I fixed it and now it uses casl to decide if user is able to update something or not.

**Review by nitrosx — APPROVED** [review]

**Pull request merged**